src/components/AccountLists/AccountLists.tsx

```diff
@@ -76,18 +76,37 @@ const AccountLists = ({ data }: Props): ReactElement => {
               ({
                 id,
                 name,
-                monthlyGoal,
+                monthlyGoal: preferencesGoal,
                 receivedPledges,
                 totalPledges,
-                currency,
+                currency: preferencesCurrency,
+                healthIndicatorData,
               }) => {
+                const monthlyGoal =
+                  preferencesGoal ?? healthIndicatorData?.machineCalculatedGoal;
+                const currency =
+                  typeof preferencesGoal === 'number'
+                    ? preferencesCurrency
+                    : healthIndicatorData?.machineCalculatedGoalCurrency;
+
+                // If the currency comes from the machine calculated goal and is different from the
+                // user's currency preference, we can't calculate the received or total percentages
+                // because the numbers are in different currencies
                 const receivedPercentage =
-                  receivedPledges / (monthlyGoal ?? NaN);
-                const totalPercentage = totalPledges / (monthlyGoal ?? NaN);
+                  currency === preferencesCurrency && monthlyGoal
+                    ? receivedPledges / monthlyGoal
+                    : NaN;
+                const totalPercentage =
+                  currency === preferencesCurrency && monthlyGoal
```

Comment:
You could make this if statement into a variable so we're don't repeat the code.